README.md

```diff
@@ -183,22 +183,71 @@ Current behavior:
 type Colors = 'red color' | 'green-color' | 'blue_color';
 ```
 
-# Development
+# Development guide
+This section provides information on how to get contributing to this project.
+## 1.- Cloning the repo
+```
+git clone --recursive https://github.com/Azure/autorest.typescript.gi
+```
+
+> ⚠️ Important:
+>
+> This project uses a git submodule for dependent code. That is why `--recursive` needs to be added when cloning this repository. If the project has been cloned already, run `git submodule update --init --recursive`.
+
+This project is not based on the default branch `main`. To visualize it, change the branch to `v4x`:
+```
+git checkout v4x
+```
+
+## 2.- Requirements
+For this project, `Dotnet 2.0`, `Gulp v3` and `Node v10` are required. Also, Linux is recommended to work on this project.
 
-### IMPORTANT NOTES
+> Node 12 can be used once Gulp is updated to v4.
```

Comment:
TBH I don't think we'll get to update gulp. I'd rather remove this note and mention that node 10 is required